pkg/deploy/deploy.go

```diff
@@ -488,4 +388,132 @@ func OperatorExists(cli client.Client, operatorPrefix string) (bool, error) {
 	return false, nil
 }
 
+func getResource(ctx context.Context, cli client.Client, obj *unstructured.Unstructured) (*unstructured.Unstructured, error) {
+	found := obj.DeepCopy()
+	err := cli.Get(ctx, types.NamespacedName{Name: obj.GetName(), Namespace: obj.GetNamespace()}, found)
+	return found, err
+}
+
+func handleDisabledComponent(ctx context.Context, cli client.Client, found *unstructured.Unstructured, componentName string, owner metav1.Object) error {
+	if found == nil {
+		return nil
+	}
+
+	resourceLabels := found.GetLabels()
+	componentCounter := getComponentCounter(resourceLabels)
+
+	if isSharedResource(componentCounter, componentName) || found.GetKind() == "CustomResourceDefinition" {
+		return nil
+	}
+
+	return updateOwnerReferencesAndDelete(ctx, cli, found, componentName, owner)
+}
+
+func getComponentCounter(foundLabels map[string]string) []string {
+	var componentCounter []string
+	for label := range foundLabels {
+		if strings.Contains(label, labels.ODHAppPrefix) {
+			compFound := strings.Split(label, "/")[1]
+			componentCounter = append(componentCounter, compFound)
+		}
+	}
+	return componentCounter
+}
+
+func isSharedResource(componentCounter []string, componentName string) bool {
+	return len(componentCounter) > 1 || (len(componentCounter) == 1 && componentCounter[0] != componentName)
+}
+
+func updateOwnerReferencesAndDelete(ctx context.Context, cli client.Client, found *unstructured.Unstructured, componentName string, owner metav1.Object) error {
+	existingOwnerReferences := found.GetOwnerReferences()
+	selector := labels.ODH.Component(componentName)
+	resourceLabels := found.GetLabels()
+
+	if resourceLabels[selector] != "true" {
+		return nil
+	}
+
+	if existingOwnerReferences == nil || hasInvalidOwnerReferences(existingOwnerReferences) {
+		return removeOwnerReferencesAndDelete(ctx, cli, found, owner)
+	}
+	return nil
+}
+
+func hasInvalidOwnerReferences(ownerReferences []metav1.OwnerReference) bool {
+	for _, owner := range ownerReferences {
+		if owner.Kind != "DataScienceCluster" && owner.Kind != "DataScienceInitialization" {
+			return false
+		}
+	}
+	return true
+}
+
+func removeOwnerReferencesAndDelete(ctx context.Context, cli client.Client, found *unstructured.Unstructured, owner metav1.Object) error {
+	found.SetOwnerReferences([]metav1.OwnerReference{})
+	data, err := json.Marshal(found)
+	if err != nil {
+		return err
+	}
+
+	if err := cli.Patch(ctx, found, client.RawPatch(types.ApplyPatchType, data), client.ForceOwnership, client.FieldOwner(owner.GetName())); err != nil {
+		return err
+	}
+
+	return cli.Delete(ctx, found)
+}
+
+func createResource(ctx context.Context, cli client.Client, obj *unstructured.Unstructured, owner metav1.Object) error {
+	if obj.GetKind() != "CustomResourceDefinition" && obj.GetKind() != "OdhDashboardConfig" {
```

Comment:
It's a change in logic with the old code, "OdhDashboardConfig" was handled here as well, right?